webpack/assets/javascripts/react_app/common/hooks/API/APIHooks.js

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { selectAPIResponse } from '../../../redux/API/APISelectors';
+import { APIActions } from '../../../redux/API';
+
+/**
+ * A custom hook that creates an API request
+ * @param  {string} key a unique key for storing the API data in redux store
```

Comment:
Thinking out loud here, do we actually need to have the user set the key? maybe we could just generate a random uuid for every request, since the consumer doesn't access the store directly but rather gets just the response?

Reply:
I wondered about it as well, though if this hook renders a random uuid the consumer won't be able to get the data in another place with the API selectors.

Reply:
Indeed, but in some cases, the caller needs the key for accessing it though I do like the simplicity of removing it.
With this update, `useAPI` will generate a uuid key, but it also returns it to the caller:
```js
  const {
    response: { results },
    key, // e9d3090a-3ff8-11eb-b378-0242ac130002
  } = useAPI('get', '/url');
```
The caller can add his own key:
```js
  const options = { key: 'SOME_KEY' };
  const {
    response: { results },
  } = useAPI('get', '/url', options);
```